cue/validate.go

```diff
@@ -0,0 +1,101 @@
+package cue
+
+import (
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/cuecontext"
+	cueerror "cuelang.org/go/cue/errors"
+	"cuelang.org/go/encoding/yaml"
+)
+
+var (
+	//go:embed flipt.cue
+	cueFile []byte
+)
+
+// ValidateBytes takes a slice of bytes, and validates them against a cue definition.
+func ValidateBytes(b []byte) error {
+	cctx := cuecontext.New()
+
+	return validateHelper(b, cctx)
+}
+
+func validateHelper(b []byte, cctx *cue.Context) error {
+	v := cctx.CompileBytes(cueFile)
+
+	f, err := yaml.Extract("", b)
+	if err != nil {
+		return err
+	}
+
+	yv := cctx.BuildFile(f, cue.Scope(v))
+	yv = v.Unify(yv)
+
+	return yv.Validate()
+}
+
+// CueError is a collection of fields that represent positions in files where the user
+// has made some kind of error.
+type CueError struct {
+	Filename string `json:"filename,omitempty"`
```

Comment:
for compatibility with `ffs` can we call this file?

Reply:
actually, the format for `ffs` is 

```
{
  message:
  location: {
    file:
    startLine:
    startColumn:
    endLine:
    endColumn:
  }
}
```

could we nest the location specific bits like that? i know that we don't have a start/end here. so maybe it could get output as:

```
{
  message:
  location: {
    file:
    line:
    column:
  }
}
```

?

Reply:
@markphelps Done! Thanks Mark.